File: laravel_api/app/Models/Restaurant.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Factories\HasFactory;

class Restaurant extends Model
{
    use HasFactory;

    protected $fillable = [

        'name',
        'address',
        'street_number',
        'postcode',
        'city',
        'province',
        'user_id'
    ];

    public function menu()
    {
        return $this->hasMany(Menu::class);
    }
    public function user()
    {
        return $this->belongsTo(User::class);
    }
}

File: menu_flutter/lib/models/app/restaurant.dart
class Restaurant {
  Restaurant({
    required this.id,
    required this.name,
    required this.address,
    required this.streetNumber,
    required this.postcode,
    required this.city,
    required this.province,
  });

  int id;
  String name;
  String address;
  String streetNumber;
  String postcode;
  String city;
  String province;

  static Restaurant fromJson(dynamic obj) {
    return Restaurant(
      id: obj['restaurant']['id'],
      name: obj['restaurant']['name'] ?? '',
      address: obj['restaurant']['address'] ?? '',
      streetNumber: obj['restaurant']['street_number'] ?? '',
      postcode: obj['restaurant']['postcode'] ?? '',
      city: obj['restaurant']['city'] ?? '',
      province: obj['restaurant']['province'] ?? '',
    );
  }
}
